File: backend/routers/upload_multi.py
from fastapi import APIRouter, UploadFile, File
import os

router = APIRouter()

@router.post("/upload/multiple")
async def upload_multiple(files: list[UploadFile] = File(...)):
    count = 0
    for file in files:
        path = os.path.join("backend/static", file.filename)
        with open(path, "wb") as f:
            f.write(await file.read())
        count += 1
    return {"message": "上传成功", "count": count}

File: frontend/js/upload_multi.js
function uploadMultiple(files) {
  const formData = new FormData();
  for (const file of files) {
    formData.append("files", file);
  }
  fetch("/upload/multiple", {
    method: "POST",
    body: formData
  }).then(res => res.json())
    .then(data => alert("上传成功: " + data.count + " 个文件"));
}
